src/nwb_conversion_tools/tools/yaml_conversion_specification/yaml_conversion_specification.py

```diff
@@ -4,14 +4,50 @@
 from importlib import import_module
 from itertools import chain
 from jsonschema import validate, RefResolver
+from typing import Optional
 
+import click
 from dandi.organize import create_unique_filenames_from_metadata
 from dandi.metadata import _get_pynwb_metadata
 
 from ...nwbconverter import NWBConverter
 from ...utils import dict_deep_update, load_dict_from_file, FilePathType, OptionalFolderPathType
 
 
+@click.command()
+@click.argument("specification-file-path")
+@click.option(
+    "--data-folder",
+    help="Folder where the source data may be found.",
+    type=click.Path(writable=True),
+)
+@click.option(
+    "--output-folder",
+    default=None,
+    help="Folder to save the output NWBFile to.",
+    type=click.Path(writable=True),
+)
+@click.option("--overwrite", help="Overwrite an existing NWBFile at the location.", is_flag=True)
+def run_conversion_from_yaml_cli(
+    specification_file_path: str,
+    data_folder: Optional[str] = None,
```

Comment:
`OptionalFolderPathType`?

Reply:
Nope, annotation types for CLI functions must be strictly strings, `None`, or occasionally `bool` if a 'flag'. It has its own system for specifying that an argument is a file/folder type of path, but does not interact with the `pathlib` library in that way.